src/java/org/apache/cassandra/db/marshal/MapType.java

```diff
@@ -77,6 +78,18 @@ private MapType(AbstractType<K> keys, AbstractType<V> values, boolean isMultiCel
         this.isMultiCell = isMultiCell;
     }
 
+    @Override
+    public MapType<K,V> overrideKeyspace(Function<String, String> overrideKeyspace)
+    {
+        List<AbstractType<?>> subTypes = subTypes();
+        AbstractType<?> newKeyType = subTypes.get(0).overrideKeyspace(overrideKeyspace);
+        AbstractType<?> newValueType = subTypes.get(1).overrideKeyspace(overrideKeyspace);
+        if (newKeyType == subTypes.get(0) && newValueType == subTypes.get(1))
+            return this;
+
+        return new MapType<>((AbstractType<K>) newKeyType, (AbstractType<V>) newValueType, isMultiCell());
```

Comment:
see https://github.com/datastax/cassandra/pull/331/files#r801303769, we should use `getInstance` to reduce GC.